### PR TITLE
[feature-wip](resource-group) support resource group interface in be.

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: #18098

## Problem summary

Describe your changes.

## Checklist(Required)

* [x] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
run buildall

File: be/src/runtime/task_group/task_group.h [hunk 1]
```diff
@@ -60,23 +66,40 @@ using TGEntityPtr = TaskGroupEntity*;
...
 public:
-    TaskGroup(uint64_t id, std::string name, uint64_t cpu_share);
+    TaskGroup(uint64_t id, std::string name, uint64_t cpu_share, int64_t version);
 
     TaskGroupEntity* task_entity() { return &_task_entity; }
 
-    uint64_t share() const { return _share; }
+    uint64_t cpu_share() const {
+        std::shared_lock<std::shared_mutex> rl {mutex};
+        return _cpu_share;
```
Comment (wangbo) [inline be/src/runtime/task_group/task_group.h:75]: How about using std::atomic for _cpu_share?
method cpu_share() will be called frequently.
Maybe just keep the eventually consistency?

**Comment by luozenglin** [comment]
run buildall

File: be/src/runtime/task_group/task_group.h (hunk 1 above)
Comment (luozenglin) [reply be/src/runtime/task_group/task_group.h]: has been replaced with atomic

**Comment by luozenglin** [comment]
run buildall

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**